### PR TITLE
Me: Remove old Gravatar upload code and clean up ProfileGravatar

### PR DESCRIPTION
**Pull request by v18**
This PR has these changes:
- remove all `me/edit-gravatar` checks
- remove the `me/edit-gravatar` flag from all environment config files
- remove unused code from `<ProfileGravatar />` and update it with these things:
  - remove debug
  - remove unneeded CSS
  - remove username link to Gravatar profile page
  - add CSS to make the plain text username look the same as before when it was a link
  - add an animation to the Gravatar image - the same one that's currently being used in production

`<ProfileGravatar />` before:

before

While hovering:

before-hover

Link:

before-link

After:

after

### CODE REVIEW

**Comment by gwwar** [comment]
Are we unblocked now by https://github.com/Automattic/wp-calypso/pull/16475?

**Comment by sirbrillig** [comment]
> Are we unblocked now by #16475?

Yup. This is deployed in staging, horizon, and wpcalypso and we haven't noticed any issues with offline uploads mis-reporting status.

File: client/me/profile-gravatar/index.jsx [hunk 1]
```diff
@@ -12,80 +11,36 @@ import { connect } from 'react-redux';
...
-
-						<span className="profile-gravatar__edit-label-wrap">
-							<span className="profile-gravatar__edit-label">
-								{ this.props.translate( 'Update Profile Photo' ) }
-							</span>
-						</span>
-					</ExternalLink>
-				</Animate>
-				<h2 className="profile-gravatar__user-display-name">{ this.props.user.display_name }</h2>
+				<div onClick={ this.handleImageClick }>
```
Comment (gwwar) [inline client/me/profile-gravatar/index.jsx]: Optional, but we can rename this to `this.recordGravatarMisclick` or something similar.

File: client/me/profile-gravatar/index.jsx [hunk 2]
```diff
@@ -12,80 +11,36 @@ import { connect } from 'react-redux';
 import Animate from 'components/animate';
 import Gravatar from 'components/gravatar';
 import { recordGoogleEvent } from 'state/analytics/actions';
-import { isEnabled } from 'config';
```
Comment (gwwar) [inline client/me/profile-gravatar/index.jsx]: I think it's probably safer to keep the config flag, so we can flip this off quickly if something bad happens, then follow up by removing this in another PR. 

If you feel pretty confident here we can run with it, but maybe keep this PR bookmarked.

Comment (sirbrillig) [inline client/me/profile-gravatar/index.jsx]: I was going to suggest something similar. It can be helpful to leave the config flag for future issues.

Comment (v18) [inline client/me/profile-gravatar/index.jsx]: Cool, opened a new PR with enabling it in production using the feature flag: https://github.com/Automattic/wp-calypso/pull/16604

**Comment by gwwar** [comment]
@v18 can we close this one out? This should already be enabled in your alternate PR.

**Comment by v18** [comment]
@gwwar definitely, thanks for the ping. I updated a function name based on a previous comment, & rebased. I tested it on Calypso & Desktop. It's ready for review again.

File: client/me/profile-gravatar/index.jsx (hunk 1 above)
Comment (v18) [reply client/me/profile-gravatar/index.jsx]: Sounds great, updated.

**Comment by v18** [comment]
Also removed feature flag in the production environment - it was added after this PR was opened. Re-tested on local & Desktop.

**Pull request merged**